types/chat.ts

```diff
@@ -122,17 +124,90 @@ export interface ChatMessage {
         text?: string
         options?: ChatItemAction[]
     }
+    followUpsHeader?: string
     codeReference?: ReferenceTrackerInformation[]
+    relatedSuggestions?: Suggestion[]
+    searchResults?: Suggestion[]
+    triggerID?: string
+    userIntent?: string
+    codeBlockLanguage?: string
     fileList?: FileList
     contextList?: FileList
+    title?: string
+    fullWidth?: boolean
+    padding?: boolean
 }
 // Response for chat prompt request can be empty,
 // if server chooses to handle the request and push updates asynchronously.
-export interface ChatResult extends ChatMessage {}
+export interface ChatResult extends ChatMessage {
+    codeBlockActions?: CodeBlockActions | null
+    buttons?: ChatItemButton[]
```

Comment:
Can we use and extend `ChatItemAction`? Otherwise just introduce new type that does not depend on MynahUI specifics.